packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -344,54 +338,70 @@ export const makeInnerVault = (
 
   /** @type {OfferHandler} */
   const closeHook = async seat => {
-    assertVaultIsOpen();
+    assertCloseable();
     assertProposalShape(seat, {
       give: { RUN: null },
       want: { Collateral: null },
     });
-    const {
-      give: { RUN: runReturned },
-      want: { Collateral: _collateralWanted },
-    } = seat.getProposal();
 
-    // you're paying off the debt, you get everything back. If you were
-    // underwater, we should have liquidated some collateral earlier: we
-    // missed our chance.
+    if (phase === VaultPhase.ACTIVE) {
+      // you're paying off the debt, you get everything back. If you were
+      // underwater, we should have liquidated some collateral earlier: we
+      // missed our chance.
+      const currentDebt = getCurrentDebt();
+      const {
+        give: { RUN: runOffered },
+      } = seat.getProposal();
 
-    // you must pay off the entire remainder but if you offer too much, we won't
-    // take more than you owe
-    const currentDebt = getCurrentDebt();
-    assert(
-      AmountMath.isGTE(runReturned, currentDebt),
-      X`You must pay off the entire debt ${runReturned} > ${currentDebt}`,
-    );
+      // you must pay off the entire remainder but if you offer too much, we won't
+      // take more than you owe
+      assert(
+        AmountMath.isGTE(runOffered, currentDebt),
+        X`Offer ${runOffered} is not sufficient to pay off debt ${currentDebt}`,
+      );
 
-    // Return any overpayment
+      // Return any overpayment
+
+      const { zcfSeat: burnSeat } = zcf.makeEmptySeatKit();
+      burnSeat.incrementBy(seat.decrementBy(harden({ RUN: currentDebt })));
+      seat.incrementBy(
+        vaultSeat.decrementBy(
+          harden({ Collateral: getCollateralAllocated(vaultSeat) }),
+        ),
+      );
+      zcf.reallocate(seat, vaultSeat, burnSeat);
+      runMint.burnLosses(harden({ RUN: currentDebt }), burnSeat);
+
+      burnSeat.exit();
+    } else if (phase === VaultPhase.LIQUIDATED) {
+      // Simply reallocate vault assets to the offer seat.
+      // Don't take anything from the offer, even if vault is underwater.
+      seat.incrementBy(
+        vaultSeat.decrementBy(
+          harden({
+            Collateral: getCollateralAllocated(vaultSeat),
+            RUN: getRunAllocated(vaultSeat),
+          }),
+        ),
+      );
```

Comment:
```suggestion
      seat.incrementBy(vaultSeat.decrementBy(vaultSeat.getCurrentAllocation()));
```

Reply:
Thanks for the code. I thought the burnSeat change is the same that I made but I must have had an error.